connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/DataTypeProtoConverter.scala

```diff
@@ -98,7 +98,7 @@ object DataTypeProtoConverter {
   }
 
   private def toCatalystStructType(t: proto.DataType.Struct): StructType = {
-    val fields = t.getFieldsList.toSeq.map { protoField =>
+    val fields = t.getFieldsList.asScala.toSeq.map { protoField =>
```

Comment:
Although it's unrelated to this PR, it might be better to use `t.getFieldsList.asScala.toArray` here. This way, we can save a collection type conversion when building the `StructType` later.